### ISSUE TITLE
mismatch with documentation for testing part 2

### ISSUE THREAD

**Issue by marker07** [mdenolle/NoisePy#29]
==> i try to run S1_fft_cc_MPI.py 

$ python3 S1_fft_cc_MPI.py

==> and then i get

File "S1_fft_cc_MPI.py", line 354
if (station[iiR]~=station[iiS]):continue
^
SyntaxError: invalid syntax

==> so i change the modelu to be :
if (station[iiR]==station[iiS]):continue

==> but i get some error again

found 41 stations in total
working on station ARV and trace bhe_00
N and Nfft are 60 (proposed 60),36000 (proposed 18000)
Traceback (most recent call last):
File "S1_fft_cc_MPI.py", line 298, in
fft_array[iii] = data.reshape(data.size)
ValueError: could not broadcast input array from shape (1080000,) into shape (540000,)

how to fix this error? thanks

**Comment by weihongpu** [comment]
Have you solved it now? I have encountered the same problem.

**Comment by mdenolle** [comment]
sorry for the delay, we are working on going through the issues now. We appreciate your patience.

**Comment by nrich20** [comment]
The issue is that the array of smaller size cannot be reshaped into a larger array. I tried the following code and it seemed to resolve the issue: 

data= data.reshape(1, data.size)
filler=np.empty((1,data.size))
fft_array[iii] = np.concatenate((data,filler),axis=1)

In this way the data array is reshaped to the 1, data.shape format and by concatenating an empty array of equal size to the end of it, the array can then be inserted into fft_array[iii]. I'm not sure how this change would affect any data processing, still conducting tests on that. This change also requires you to make additional changes in the code down the line as parts of the code try to do the reverse of the original function and are unable to do that. 

For example, line 323: 
 sfft1 = sfft1.reshape(N,Nfft2)
This line tries to reshape sfft1 into an array that is half its size and numpy won't allow that. To change this, I implemented the following:

sfft1 = noise_module.smooth_source_spect(fc_para,fft1)
sfft1_halved= sfft1[0:(int(sfft1.size/2))]
sfft1 = sfft1_halved.reshape(N,Nfft2)

The idea being that the last half of the array is the empty spaces that I inserted in my earlier solution to do the reshaping, so removing the last half would not remove any parts of the data. So, then this code segment reduces sfft1 to the proper size for reshaping and then does so. 

Something similar happens on line 358 as well: 
  fft2 = fft_array[iiR];sfft2 = fft2.reshape(N,Nfft2)

I implemented the following to resolve this problem: 
fft2 = fft_array[iiR]
fft2_halved= fft2[0:(int(fft2.size/2))]
sfft2 = fft2_halved.reshape(N,Nfft2)
 
I still have to do more testing to see if the data is impacted somehow by these solutions, will update if I find a problem.
